src/main/java/org/apache/hadoop/mapred/ResourcePolicy.java

```diff
@@ -383,16 +383,22 @@ public void resourceOffers(SchedulerDriver schedulerDriver,
         }
 
         String uri = scheduler.conf.get("mapred.mesos.executor.uri");
-        if (uri == null) {
+        String directory = scheduler.conf.get("mapred.mesos.executor.directory");
+        boolean isUriSet = uri != null && !uri.equals("");
+        boolean isDirectorySet = directory != null && !directory.equals("");
+
+        if (!isUriSet && !isDirectorySet) {
           throw new RuntimeException(
               "Expecting configuration property 'mapred.mesos.executor'");
-        }
-
-        String directory = scheduler.conf.get("mapred.mesos.executor.directory");
-        if (directory == null || directory.equals("")) {
+        } else if (isUriSet && isDirectorySet) {
+          throw new RuntimeException(
+              "Conflicting properties 'mapred.mesos.executor.uri' and 'mapred.mesos.executor.directory', only one can be set");
```

Comment:
hmm, so that would mean we'd have to check if the directory property represents a relative or absolute path.
All in all, I think this is getting messy by reusing the directory property (which is my fault :) ). Maybe we should rethink this. Do you have any ideas? Or just not throw the exception and a useful log entry?